tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -186,19 +218,15 @@ goto node = do
   cursor <- ask
   liftIO (with node (ts_tree_cursor_reset_p cursor))
 
--- | Return the 'Node' that the cursor is pointing at (if any), or 'Nothing' otherwise.
-peekNode :: (Carrier sig m, Member (Reader (Ptr Cursor)) sig, MonadIO m) => m (Maybe Node)
+-- | Return the 'Node' that the cursor is pointing at.
+peekNode :: (Carrier sig m, Member (Reader (Ptr Cursor)) sig, MonadIO m) => m Node
```

Comment:
When originally written, we had used API which walked the tree using the next sibling API, which indicates that it’s reached the end by returning a node for which `ts_node_is_null` returns true. By contrast, the cursor API, which we now rely on exclusively, is guarded: API to change the cursor returns a boolean value indicating whether the change occurred. Thus, the cursor API won’t return null nodes, and we don’t have to return in `Maybe`, enabling further tidying.